diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7401224..97388fd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -361,11 +361,6 @@ endif (_usebison STREQUAL YES)
 if (_uselapack STREQUAL YES)
     find_package (BLAS REQUIRED)
     find_package (LAPACK REQUIRED)
-    if (FFTW3_DISABLE_THREADS)
-        find_package (FFTW3 COMPONENTS single double REQUIRED)
-    else()
-        find_package (FFTW3 COMPONENTS single double threads REQUIRED)
-    endif (FFTW3_DISABLE_THREADS)
 endif (_uselapack STREQUAL YES)
 if (_usefits STREQUAL YES)
     find_package (CFITSIO 3.030 REQUIRED) # Should pad to three decimal digits
@@ -410,11 +405,15 @@ if (HDF5_FOUND)
     add_definitions(-DHAVE_HDF5)
 endif (HDF5_FOUND)
 
-include_directories (SYSTEM ${FFTW3_INCLUDE_DIRS})
-add_definitions(-DHAVE_FFTW3)
-if (NOT FFTW3_DISABLE_THREADS)
-    add_definitions(-DHAVE_FFTW3_THREADS)
-endif (NOT FFTW3_DISABLE_THREADS)
+if (BUILD_FFTW3)
+    include_directories (SYSTEM ${FFTW3_INCLUDE_DIRS})
+    add_definitions(-DHAVE_FFTW3)
+    if (NOT FFTW3_DISABLE_THREADS)
+        add_definitions(-DHAVE_FFTW3_THREADS)
+    endif (NOT FFTW3_DISABLE_THREADS)
+else()
+    remove_definitions((-DHAVE_FFTW3 -DHAVE_FFTW3_THREADS))
+endif (BUILD_FFTW3)
 
 if (DL_FOUND)
     add_definitions(-DHAVE_DL)
@@ -567,8 +566,41 @@ endforeach (module)
 
 # Install pkg-config support file
 CONFIGURE_FILE("casacore.pc.in" "casacore.pc" @ONLY)
-set(CASA_PKGCONFIG_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig")
-INSTALL(FILES "${CMAKE_CURRENT_BINARY_DIR}/casacore.pc" DESTINATION "${CASA_PKGCONFIG_INSTALL_PREFIX}")
+# set(CASA_PKGCONFIG_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig")
+# INSTALL(FILES "${CMAKE_CURRENT_BINARY_DIR}/casacore.pc" DESTINATION "${CASA_PKGCONFIG_INSTALL_PREFIX}")
+INSTALL(FILES "${CMAKE_CURRENT_BINARY_DIR}/casacore.pc" DESTINATION lib/pkgconfig)
+
+# generate and install export file
+install(EXPORT casacore
+    FILE "${PROJECT_NAME}Targets.cmake"
+    NAMESPACE ${namespace}::
+    DESTINATION cmake
+)
+
+include(CMakePackageConfigHelpers)
+
+# generate the version file for the config file
+write_basic_package_version_file(
+    "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+    VERSION "${version}"
+    COMPATIBILITY AnyNewerVersion
+)
+# create config file
+configure_package_config_file(${CMAKE_CURRENT_SOURCE_DIR}/Config.cmake.in
+    "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+    INSTALL_DESTINATION cmake
+)
+# install config files
+install(FILES
+    "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+    "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+    DESTINATION cmake
+)
+# generate the export targets for the build tree
+export(EXPORT casacore
+    FILE "${CMAKE_CURRENT_BINARY_DIR}/cmake/${PROJECT_NAME}Targets.cmake"
+    NAMESPACE ${namespace}::
+)
 
 # Show summary.
 message (STATUS "CMAKE_SYSTEM .......... = ${CMAKE_SYSTEM}")
diff --git a/Config.cmake.in b/Config.cmake.in
new file mode 100644
index 0000000..8c9ad12
--- /dev/null
+++ b/Config.cmake.in
@@ -0,0 +1,5 @@
+@PACKAGE_INIT@
+
+include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@Targets.cmake")
+
+check_required_components(@PROJECT_NAME@)
diff --git a/casa/CMakeLists.txt b/casa/CMakeLists.txt
index 60fa2b6..a573d61 100644
--- a/casa/CMakeLists.txt
+++ b/casa/CMakeLists.txt
@@ -15,7 +15,9 @@ set (
 parser_inputs
 JsonGram
 )
- 
+
+find_package(ZLIB REQUIRED)
+
 foreach (src ${parser_inputs})
     if (BISON_VERSION VERSION_LESS 3.0)
         set(EXTRA_BISON_FLAGS "-p ${src}")
@@ -294,16 +296,18 @@ find_library(libm m)
 
 target_link_libraries (
 casa_casa
+PUBLIC
 ${de_libraries}
 ${libm}
 dl
 ${CASACORE_ARCH_LIBS}
+PRIVATE
+ZLIB::ZLIB
 )
 
-add_subdirectory (apps)
-
 install (
 TARGETS casa_casa
+EXPORT casacore
 LIBRARY DESTINATION lib${LIB_SUFFIX}
 ARCHIVE DESTINATION lib${LIB_SUFFIX}
 LIBRARY PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
diff --git a/casa/Exceptions/Error.h b/casa/Exceptions/Error.h
index 12229f9..a15e508 100644
--- a/casa/Exceptions/Error.h
+++ b/casa/Exceptions/Error.h
@@ -91,7 +91,7 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
 #define ThrowIfError(c,m) {if (AIPS_UNLIKELY(c)) {casacore::AipsError::throwIfError (casacore::True, (m), __FILE__, __LINE__, __PRETTY_FUNCTION__);}}
 
 // Repackage and rethrow an AipsError exception.
-#define Rethrow(e,m) {throw casacore::AipsError::repackageAipsError ((e),(m),__FILE__,__LINE__, __PRETTY_FUNCTION__);}
+// #define Rethrow(e,m) {throw casacore::AipsError::repackageAipsError ((e),(m),__FILE__,__LINE__, __PRETTY_FUNCTION__);}
 
 
 // <summary>Base class for all Casacore library errors</summary>
diff --git a/casa/Json/JsonOut.h b/casa/Json/JsonOut.h
index b5229c5..7250d93 100644
--- a/casa/Json/JsonOut.h
+++ b/casa/Json/JsonOut.h
@@ -174,6 +174,10 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
     void put (const DComplex& value);
     void put (const char* value);
     void put (const String& value);
+    // Put a Record which is written as a {} structure.
+    // The Record can be nested.
+    void put (const Record&);
+
     // </group>
 
     // Put a line defining an array value. Multi-dim arrays are written as
@@ -215,10 +219,6 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
     // Write a key and valueholder.
     void writeKV (const String& name, const ValueHolder& vh);
 
-    // Put a Record which is written as a {} structure.
-    // The Record can be nested.
-    void put (const Record&);
-
     // Get the indentation after a name.
     // It indents with the length of the name (including quotes and colon)
     // with a maximum of 20 spaces.
diff --git a/cmake/FindFFTW3.cmake b/cmake/FindFFTW3.cmake
index 886f3be..66d1176 100644
--- a/cmake/FindFFTW3.cmake
+++ b/cmake/FindFFTW3.cmake
@@ -58,15 +58,6 @@ foreach(_comp ${_components})
   endif(_comp STREQUAL "single")
 endforeach(_comp ${_components})
 
-# If using threads, we need to link against threaded libraries as well.
-if(_use_threads)
-  set(_thread_libs)
-  foreach(_lib ${_libraries})
-    list(APPEND _thread_libs ${_lib}_threads)
-  endforeach(_lib ${_libraries})
-  set(_libraries ${_thread_libs} ${_libraries})
-endif(_use_threads)
-
 # Keep a list of variable names that we need to pass on to
 # find_package_handle_standard_args().
 set(_check_list)
diff --git a/coordinates/CMakeLists.txt b/coordinates/CMakeLists.txt
index 783ac1d..5871327 100644
--- a/coordinates/CMakeLists.txt
+++ b/coordinates/CMakeLists.txt
@@ -31,6 +31,7 @@ init_pch_support(casa_coordinates ${top_level_headers})
 target_link_libraries (casa_coordinates casa_fits ${WCSLIB_LIBRARIES} ${CASACORE_ARCH_LIBS})
 
 install (TARGETS casa_coordinates
+EXPORT casacore
 LIBRARY DESTINATION lib${LIB_SUFFIX}
 ARCHIVE DESTINATION lib${LIB_SUFFIX}
 LIBRARY PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
diff --git a/derivedmscal/CMakeLists.txt b/derivedmscal/CMakeLists.txt
index ff8da2a..232e5b9 100644
--- a/derivedmscal/CMakeLists.txt
+++ b/derivedmscal/CMakeLists.txt
@@ -19,6 +19,7 @@ init_pch_support(casa_derivedmscal ${top_level_headers})
 target_link_libraries (casa_derivedmscal casa_ms ${CASACORE_ARCH_LIBS})
 
 install (TARGETS casa_derivedmscal
+EXPORT casacore
 LIBRARY DESTINATION lib${LIB_SUFFIX}
 ARCHIVE DESTINATION lib${LIB_SUFFIX}
 LIBRARY PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
diff --git a/fits/CMakeLists.txt b/fits/CMakeLists.txt
index 65b888e..47cf365 100644
--- a/fits/CMakeLists.txt
+++ b/fits/CMakeLists.txt
@@ -30,14 +30,19 @@ FITS.h
 
 init_pch_support(casa_fits ${top_level_headers})
 
-target_link_libraries (casa_fits casa_measures ${CFITSIO_LIBRARIES} ${CASACORE_ARCH_LIBS})
+find_package(ZLIB REQUIRED)
 
-add_subdirectory (apps)
+target_link_libraries (casa_fits
+    PUBLIC casa_measures ${CFITSIO_LIBRARIES} ${CASACORE_ARCH_LIBS}
+    PRIVATE ZLIB::ZLIB)
+
+# apps subdirectory disabled (arcae static monolith build)
 
 #the tests
 add_subdirectory (FITS/test ${EXCL_ALL})
 
 install (TARGETS casa_fits
+EXPORT casacore
 LIBRARY DESTINATION lib${LIB_SUFFIX}
 ARCHIVE DESTINATION lib${LIB_SUFFIX}
 LIBRARY PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
diff --git a/fits/apps/CMakeLists.txt b/fits/apps/CMakeLists.txt
index b2e4220..45dde2e 100644
--- a/fits/apps/CMakeLists.txt
+++ b/fits/apps/CMakeLists.txt
@@ -1,4 +1,5 @@
+find_package(ZLIB REQUIRED)
 add_executable (fits2table  fits2table/fits2table.cc)
 add_pch_support(fits2table)
-target_link_libraries (fits2table casa_fits ${CASACORE_ARCH_LIBS})
+target_link_libraries (fits2table PUBLIC casa_fits ${CASACORE_ARCH_LIBS} PRIVATE ZLIB::ZLIB)
 install(TARGETS fits2table)
diff --git a/images/CMakeLists.txt b/images/CMakeLists.txt
index 5ff84c3..4602bb1 100644
--- a/images/CMakeLists.txt
+++ b/images/CMakeLists.txt
@@ -83,10 +83,11 @@ casa_lattices
 ${CASACORE_ARCH_LIBS}
 )
 
-add_subdirectory (apps)
+# apps subdirectory disabled (arcae static monolith build)
 
 install (
 TARGETS casa_images
+EXPORT casacore
 LIBRARY DESTINATION lib${LIB_SUFFIX}
 ARCHIVE DESTINATION lib${LIB_SUFFIX}
 LIBRARY PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
diff --git a/lattices/CMakeLists.txt b/lattices/CMakeLists.txt
index dccca16..29465d7 100644
--- a/lattices/CMakeLists.txt
+++ b/lattices/CMakeLists.txt
@@ -77,6 +77,7 @@ ${CASACORE_ARCH_LIBS}
 )
 
 install (TARGETS casa_lattices
+EXPORT casacore
 LIBRARY DESTINATION lib${LIB_SUFFIX}
 ARCHIVE DESTINATION lib${LIB_SUFFIX}
 LIBRARY PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
diff --git a/meas/CMakeLists.txt b/meas/CMakeLists.txt
index a2c9d56..d3b5b65 100644
--- a/meas/CMakeLists.txt
+++ b/meas/CMakeLists.txt
@@ -30,6 +30,7 @@ init_pch_support(casa_meas ${top_level_headers})
 target_link_libraries (casa_meas casa_measures)
 
 install (TARGETS casa_meas
+EXPORT casacore
 RUNTIME DESTINATION bin
 LIBRARY DESTINATION lib${LIB_SUFFIX}
 ARCHIVE DESTINATION lib${LIB_SUFFIX}
diff --git a/measures/CMakeLists.txt b/measures/CMakeLists.txt
index 91285e3..696a859 100644
--- a/measures/CMakeLists.txt
+++ b/measures/CMakeLists.txt
@@ -78,10 +78,11 @@ init_pch_support(casa_measures ${top_level_headers})
 
 target_link_libraries (casa_measures casa_tables casa_scimath ${CASACORE_ARCH_LIBS})
 
-add_subdirectory (apps)
+# apps subdirectory disabled (arcae static monolith build)
 
 install (
 TARGETS casa_measures
+EXPORT casacore
 LIBRARY DESTINATION lib${LIB_SUFFIX}
 ARCHIVE DESTINATION lib${LIB_SUFFIX}
 LIBRARY PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
diff --git a/mirlib/CMakeLists.txt b/mirlib/CMakeLists.txt
index b8b6aac..0b71077 100644
--- a/mirlib/CMakeLists.txt
+++ b/mirlib/CMakeLists.txt
@@ -27,6 +27,7 @@ target_link_libraries (casa_mirlib casa_casa ${CASACORE_ARCH_LIBS})
 
 install (
 TARGETS casa_mirlib
+EXPORT casacore
 RUNTIME DESTINATION bin
 LIBRARY DESTINATION lib${LIB_SUFFIX}
 ARCHIVE DESTINATION lib${LIB_SUFFIX}
diff --git a/ms/CMakeLists.txt b/ms/CMakeLists.txt
index 850ab87..9ba0db0 100644
--- a/ms/CMakeLists.txt
+++ b/ms/CMakeLists.txt
@@ -189,9 +189,10 @@ casa_ms casa_measures
 ${CASACORE_ARCH_LIBS}
 )
 
-add_subdirectory (apps)
+# apps subdirectory disabled (arcae static monolith build)
 
 install (TARGETS casa_ms
+EXPORT casacore
 RUNTIME DESTINATION bin
 LIBRARY DESTINATION lib${LIB_SUFFIX}
 ARCHIVE DESTINATION lib${LIB_SUFFIX}
diff --git a/ms/MeasurementSets/MSTableImpl.cc b/ms/MeasurementSets/MSTableImpl.cc
index a5d3c4d..bb73504 100644
--- a/ms/MeasurementSets/MSTableImpl.cc
+++ b/ms/MeasurementSets/MSTableImpl.cc
@@ -321,9 +321,10 @@ void MSTableImpl::addKeyToDesc(TableDesc& td, const String& keyName,
 //#	    keywordStandardComment(key);
 	break; 
     default:
-      cerr << "Data type: "<< keyDType << ", "<< keyName<< "not handled"<<endl;
+      // cerr << "Data type: "<< keyDType << ", "<< keyName<< "not handled"<<endl;
       //	throw(AipsError ("MSTableImpl::addKeyToDesc(...) - "
       //			 "Data type not handled"));
+			break;
     }  
 }
 
diff --git a/msfits/CMakeLists.txt b/msfits/CMakeLists.txt
index b8340e0..6b434d3 100644
--- a/msfits/CMakeLists.txt
+++ b/msfits/CMakeLists.txt
@@ -32,10 +32,11 @@ init_pch_support(casa_msfits ${top_level_headers})
 
 target_link_libraries (casa_msfits casa_ms casa_fits ${CASACORE_ARCH_LIBS})
 
-add_subdirectory (apps)
+# apps subdirectory disabled (arcae static monolith build)
 
 install (
 TARGETS casa_msfits
+EXPORT casacore
 LIBRARY DESTINATION lib${LIB_SUFFIX}
 ARCHIVE DESTINATION lib${LIB_SUFFIX}
 LIBRARY PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
diff --git a/scimath/CMakeLists.txt b/scimath/CMakeLists.txt
index b7f2900..9999811 100644
--- a/scimath/CMakeLists.txt
+++ b/scimath/CMakeLists.txt
@@ -62,6 +62,7 @@ else (FFTW3_FOUND)
 endif (FFTW3_FOUND)
 
 install (TARGETS casa_scimath
+EXPORT casacore
 LIBRARY DESTINATION lib${LIB_SUFFIX}
 ARCHIVE DESTINATION lib${LIB_SUFFIX}
 LIBRARY PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
diff --git a/scimath_f/CMakeLists.txt b/scimath_f/CMakeLists.txt
index 69bd4d7..8e6aa8d 100644
--- a/scimath_f/CMakeLists.txt
+++ b/scimath_f/CMakeLists.txt
@@ -50,6 +50,7 @@ target_link_libraries (casa_scimath_f casa_tables ${LAPACK_LIBRARIES} ${BLAS_LIB
 
 install (
 TARGETS casa_scimath_f
+EXPORT casacore
 LIBRARY DESTINATION lib${LIB_SUFFIX}
 ARCHIVE DESTINATION lib${LIB_SUFFIX}
 LIBRARY PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
diff --git a/tables/CMakeLists.txt b/tables/CMakeLists.txt
index 59feaec..cd87055 100644
--- a/tables/CMakeLists.txt
+++ b/tables/CMakeLists.txt
@@ -258,15 +258,22 @@ TaQL.h
 
 init_pch_support(casa_tables ${top_level_headers})
 
-target_link_libraries (casa_tables casa_casa ${CASACORE_ARCH_LIBS} ${DYSCOSTMAN_LIBRARIES} ${CASACORE_ADIOS_LIBRARY} ${CASACORE_MPI_LIBRARY})
+target_link_libraries (casa_tables casa_casa ${CASACORE_ARCH_LIBS} ${DYSCOSTMAN_LIBRARIES})
+if(MPI_FOUND)
+  target_link_libraries(casa_tables ${CASACORE_MPI_LIBRARY})
+  if(ADIOS2_FOUND)
+    target_link_libraries(casa_tables adios2::adios2)
+  endif(ADIOS2_FOUND)
+endif(MPI_FOUND)
 if(BUILD_SISCO)
   target_link_libraries (casa_tables ${LibDeflate_LIBRARY})
   target_include_directories(casa_tables PUBLIC ${LibDeflate_INCLUDE_DIR})
 endif(BUILD_SISCO)
 
-add_subdirectory (apps)
+# apps subdirectory disabled (arcae static monolith build)
 
 install (TARGETS casa_tables
+EXPORT casacore
 LIBRARY DESTINATION lib${LIB_SUFFIX}
 ARCHIVE DESTINATION lib${LIB_SUFFIX}
 LIBRARY PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
diff --git a/tables/Tables/PlainTable.cc b/tables/Tables/PlainTable.cc
index f8f7653..fe1c079 100644
--- a/tables/Tables/PlainTable.cc
+++ b/tables/Tables/PlainTable.cc
@@ -45,7 +45,7 @@
 namespace casacore { //# NAMESPACE CASACORE - BEGIN
 
 //# Initialize the static TableCache object.
-TableCache PlainTable::theirTableCache;
+thread_local TableCache PlainTable::theirTableCache;
 
 PlainTable::PlainTable (SetupNewTable& newtab, rownr_t nrrow, Bool initialize,
                         const TableLock& lockOptions, int endianFormat,
diff --git a/tables/Tables/PlainTable.h b/tables/Tables/PlainTable.h
index 5742020..c5c5f8e 100644
--- a/tables/Tables/PlainTable.h
+++ b/tables/Tables/PlainTable.h
@@ -316,7 +316,7 @@ private:
     TSMOption      tsmOption_p;
     Bool           changeTiledDataOnly_; //# Allow updates to data in existing tiled columns
     //# cache of open (plain) tables
-    static TableCache theirTableCache;
+    static thread_local TableCache theirTableCache;
 };
 
 
diff --git a/tables/Tables/RefRows.cc b/tables/Tables/RefRows.cc
index 957b3e1..0d57066 100644
--- a/tables/Tables/RefRows.cc
+++ b/tables/Tables/RefRows.cc
@@ -38,17 +38,24 @@ RefRows::RefRows (const Vector<rownr_t>& rowNumbers, Bool isSliced,
   init (rowNumbers, isSliced, collapse);
 }
 
+RefRows::RefRows (Vector<rownr_t>&& rowNumbers, Bool isSliced,
+		  Bool collapse)
+{
+  init (std::move(rowNumbers), isSliced, collapse);
+}
+
+
 RefRows::RefRows (const Vector<uInt>& rowNumbers, Bool isSliced,
 		  Bool collapse)
 {
   init (RowNumbers(rowNumbers), isSliced, collapse);
 }
   
-void RefRows::init (const Vector<rownr_t>& rowNumbers, Bool isSliced,
+void RefRows::init (Vector<rownr_t> rowNumbers, Bool isSliced,
                     Bool collapse)
 {
-  itsRows   = rowNumbers;
-  itsNrows  = rowNumbers.nelements();
+  itsRows   = std::move(rowNumbers);
+  itsNrows  = itsRows.nelements();
   itsSliced = isSliced;
     if (itsSliced) {
 	AlwaysAssert (itsNrows%3 == 0, AipsError);
diff --git a/tables/Tables/RefRows.h b/tables/Tables/RefRows.h
index 32ab695..f1f738e 100644
--- a/tables/Tables/RefRows.h
+++ b/tables/Tables/RefRows.h
@@ -92,6 +92,9 @@ public:
     // individual row numbers to the slice form (to save memory).
     RefRows (const Vector<rownr_t>& rowNumbers, Bool isSliced = False,
              Bool collapse = False);
+
+    RefRows (Vector<rownr_t>&& rowNumbers, Bool isSliced = False,
+             Bool collapse = False);
 #ifdef IMPLICIT_CTDS_32BIT
     RefRows (const Vector<uInt>& rowNumbers, Bool isSliced = False,
              Bool collapse = False);
@@ -148,7 +151,7 @@ public:
 
 private:
     // Initialize the object.
-    void init (const Vector<rownr_t>& rowNumbers, Bool isSliced,
+    void init (Vector<rownr_t> rowNumbers, Bool isSliced,
                Bool collapse);
 
     // Fill the itsNrows variable.
diff --git a/tables/Tables/TableProxy.cc b/tables/Tables/TableProxy.cc
index f65cf26..3f806cd 100644
--- a/tables/Tables/TableProxy.cc
+++ b/tables/Tables/TableProxy.cc
@@ -271,6 +271,9 @@ Record TableProxy::lockOptions()
       option = "autonoread";
     }
     break;
+  case TableLock::NoLocking:
+    option = "nolock";
+    break;
   default:
     option = "unknown";
   }
@@ -3403,10 +3406,12 @@ TableLock TableProxy::makeLockOptions (const Record& options)
     opt = TableLock::PermanentLocking;
   } else if (str == "permanentwait") {
     opt = TableLock::PermanentLockingWait;
+  } else if (str == "nolock") {
+    opt = TableLock::NoLocking;
   } else {
     throw TableError ("'" + str + "' is an unknown lock option; valid are "
 		      "default,auto,autonoread,user,usernoread,permanent,"
-		      "permanentwait");
+		      "permanentwait,lock");
   }
   if (options.nfields() == 1) {
     return TableLock(opt);
